File: app/src/main/java/com/sparkathon/shopmate/datamodals/Product.kt
data class Rating(
    val rate: Float,
    val count: Int
)

data class Product(
    val id: Int,
    val title: String,
    val price: Double,
    val image: String,
    var quantity: Int = 0,
    val description: String,
    val category: String,
    val rating: Rating
)

File: server/models/Product.js
const mongoose = require('mongoose');

const RatingSchema = new mongoose.Schema({
    rate: Number,
    count: Number,
});

const ProductSchema = new mongoose.Schema({
    id: { type: Number, unique: true, required: true }, 
    title: { type: String, required: true },
    price: { type: Number, required: true },
    description: { type: String, required: true },
    category: { type: String, required: true },
    image: { type: String, required: true },
    rating: RatingSchema,
});

module.exports = mongoose.model('Product', ProductSchema);